pkg/cmd/pubsub.go

```diff
@@ -0,0 +1,110 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/gitops-tools/image-updater/pkg/applier"
+	"github.com/gitops-tools/image-updater/pkg/config"
+	"github.com/gitops-tools/image-updater/pkg/hooks/gcr"
+	"github.com/gitops-tools/image-updater/pkg/pubsubhandler"
+	"github.com/gitops-tools/pkg/client"
+	"github.com/go-logr/zapr"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+const (
+	projectIDFlag        = "project-id"
+	subscriptionNameFlag = "subscription-name"
+)
+
+type message struct {
+	data []byte
+}
+
+func (m *message) Ack()         {}
+func (m *message) Data() []byte { return m.data }
+
+func makePubsubCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "pubsub",
+		Short: "update repositories in response to gcr pubsub events",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			zapl, _ := zap.NewProduction()
```

Comment:
I wonder if some of this code which is duplicated from the http command could be extracted and reused?